core/src/util/builders.cpp

```diff
@@ -383,7 +372,7 @@ void Builders::buildPolyLine(const Line& _line, PolyLineBuilder& _ctx) {
         for (size_t i = 0; i < _line.size() - 1; i++) {
             const glm::vec3& coordCurr = _line[i];
             const glm::vec3& coordNext = _line[i+1];
-            if (isOnTileEdge(coordCurr, coordNext)) {
+            if (isOutsideTile(coordCurr, coordNext)) {
                 Line line = Line(&_line[cut], &_line[i+1]);
                 buildPolyLineSegment(line, _ctx);
```

Comment:
@tallytalwar ```i+1``` should be clear enough. just nitpicking. using ```some_iterator + length``` looks to me to be explicitly an end iterator.

```
       auto cut = _line.begin();
       const glm::vec3& coordCurr = _line.front();
        for (auto it = cut + 1; it < _line.end(); it++) {
              const glm::vec3& coordNext = *it;
              if (isOutsideTile(coordCurr, coordNext)) {
                  size_t length = it - cut;
                  if (length > 1) {
                      Line line = Line(cut, cut + length);
                      buildPolyLineSegment(line, _ctx);
                  }
                  cut = it;
              }
              coordCurr = coordNext;
         }
         Line line = Line(cut, _line.end());
         buildPolyLineSegment(line, _ctx);
```